### PR TITLE
Move Gen AI toggle tab state to localStorage

### PR DESCRIPTION
**Pull request by adamjarling**
## What does this do?
- Defaults the active tab on Search page to show Chat streaming response when a user enters the Search route with the Gen AI checkbox checked.
- Moves Gen AI checkbox state from URL query params, to `localStorage`
- Removes active tab state from Search Context
- Attempts to handle faceting/filtering UX when Gen AI checkbox state is active

This PR moves the ball down the field, however there is one last piece which I'm not sure of the proper direction?

When a user is search w/ Gen AI, switches to the View Results tab, actively filters the results, and then clicks on the "AI Response" tab, the AI stream creates a new websocket request and doesn't render the cached result.   Is this an expected outcome or no?  Does faceting or filtering items change the nature of the original AI response, I guess is the question.

## How to test
Basically just click around the app, searching, toggling the Gen AI check box on off at various places and times.  Try logging in / out, see how the UX responds.  Anything which seems out of the ordinary should be noted and addressed.

### CODE REVIEW

**Review by mathewjordan — CHANGES_REQUESTED** [review]
This works great aside from one point of complexity. What should the user expect when the do the following?

1. Check _Use Generative AI_
2. Do their query/question, "Music"
3. Click _View `n` Results_ tab
4. Select any facet (Image)

Right now, this will select the facet however it will also send them back to the AI Response tab. I think this is where we keep falling back an forth. This bit of complexity makes it where we can't necessarily rely on the Checkbox solely for deciding the current tab -- but some other current state knowledge is necessary as well.

**Comment by adamjarling** [comment]
> This works great aside from one point of complexity. What should the user expect when the do the following?
> 
> 1. Check _Use Generative AI_
> 2. Do their query/question, "Music"
> 3. Click _View `n` Results_ tab
> 4. Select any facet (Image)
> 
> Right now, this will select the facet however it will also send them back to the AI Response tab. I think this is where we keep falling back an forth. This bit of complexity makes it where we can't necessarily rely on the Checkbox solely for deciding the current tab -- but some other current state knowledge is necessary as well.

@mathewjordan I've been really trying to avoid suggested browser `localStorage` to track whether a user has opted into Gen AI, but your point above aligns with some other scenarios which might be pointing in that direction.  We can't really store it in top level Context (page refresh wipes it), seems clunky to force it into a URL param on every route...  Thoughts?  I can do a quick test and gauge how big a refactor this might be.

**Comment by mathewjordan** [comment]
> > This works great aside from one point of complexity. What should the user expect when the do the following?
> > 
> > 1. Check _Use Generative AI_
> > 2. Do their query/question, "Music"
> > 3. Click _View `n` Results_ tab
> > 4. Select any facet (Image)
> > 
> > Right now, this will select the facet however it will also send them back to the AI Response tab. I think this is where we keep falling back an forth. This bit of complexity makes it where we can't necessarily rely on the Checkbox solely for deciding the current tab -- but some other current state knowledge is necessary as well.
> 
> @mathewjordan I've been really trying to avoid suggested browser `localStorage` to track whether a user has opted into Gen AI, but your point above aligns with some other scenarios which might be pointing in that direction. We can't really store it in top level Context (page refresh wipes it), seems clunky to force it into a URL param on every route... Thoughts? I can do a quick test and gauge how big a refactor this might be.

I'm all for it if it simplifies handling.

**Review by mathewjordan — COMMENTED** [review]
I'm not sure if I'm missing something. I see the `ai` key in local storage now, however it doesn't seem to be defaulting to the AI Response as expected. Maybe we can check it out together?

image

**Review by mathewjordan — APPROVED** [review]
💯
